src/twisted/mail/test/test_mail.py

```diff
@@ -247,25 +248,80 @@ def _cbFinalName(self, result):
         self.assertTrue(self.f.closed)
         self.assertFalse(os.path.exists(self.name))
 
-    @skipIf(sys.version_info >= (3,), "not ported to Python 3")
     def testContents(self):
-        contents = "first line\nsecond line\nthird line\n"
+        contents = b"first line\nsecond line\nthird line\n"
         for line in contents.splitlines():
             self.fp.lineReceived(line)
         self.fp.eomReceived()
-        with open(self.final) as f:
+        with open(self.final, "rb") as f:
             self.assertEqual(f.read(), contents)
 
-    @skipIf(sys.version_info >= (3,), "not ported to Python 3")
     def testInterrupted(self):
-        contents = "first line\nsecond line\n"
+        contents = b"first line\nsecond line\n"
         for line in contents.splitlines():
             self.fp.lineReceived(line)
         self.fp.connectionLost()
         self.assertFalse(os.path.exists(self.name))
         self.assertFalse(os.path.exists(self.final))
 
 
+@skipIf(platformType != "posix", "twisted.mail only works on posix")
+class MaildirMessageTests(TestCase):
+    def setUp(self):
+        self.name = "maildirMessage.testFile"
+        self.final = "final.maildirMessage.testFile"
+        self.address = b"user@example.com"
+        self.f = open(self.name, "wb")
+        self.fp = mail.maildir.MaildirMessage(
+            self.address, self.f, self.name, self.final
+        )
+
+    def _finalName(self):
+        return glob.glob(f"{self.final},S=[0-9]*")[0]
+
+    def tearDown(self):
+        try:
+            self.f.close()
+        except BaseException:
+            pass
+        try:
+            os.remove(self.name)
+        except BaseException:
```

Comment:
Please _don't_ catch `BaseException` like this unless there's a specific need. (I don't see one here.) This construct will catch flow control exceptions like `SystemExit` which you don't want. I think that `except Exception` (or better yet, the specific exception you expect) would be more appropriate.